misc/discv5/src/query_pool/peers/closest.rs

```diff
@@ -215,6 +176,24 @@ where
                         peer.state = QueryPeerState::PendingIteration;
                     }
                 }
+                QueryPeerState::Unresponsive => {
+                    let peer = e.get_mut();
+                    peer.peers_returned += num_closest;
```

Comment:
I think we should set this to be:
```rust
peer.peers_returned += closer_peers.len();
```

This broke some of our tests last time. We may need to investigate why it breaks those tests, as I believe this is the correct logic.

Reply:
Will open another PR to fix this to keep it logically separate.